library/core/src/ptr/alignment.rs

```diff
@@ -18,7 +18,8 @@ pub struct Alignment(AlignmentEnum);
 const _: () = assert!(mem::size_of::<Alignment>() == mem::size_of::<usize>());
 const _: () = assert!(mem::align_of::<Alignment>() == mem::align_of::<usize>());
 
-fn _alignment_can_be_structurally_matched(a: Alignment) -> bool {
+#[inline]
```

Comment:
Why `#[inline]`? Does this function even get called anywhere?

Reply:
I did a quick search and it doesn't seem to be used, so i have removed it for now.

Reply:
Don't remove the entire function. It tests that `Alignment` can be structurally `match`ed, and if it can't, a compiler error is raised.

Reply:
i did not realize that, i have added it back.